src/Diagnostics.RuntimeHost/Controllers/DiagnosticControllerBase.cs

```diff
@@ -61,6 +61,55 @@ public DiagnosticControllerBase(IStampService stampService, ICompilerHostClient
             return detectorResponse == null ? (IActionResult)NotFound() : Ok(DiagnosticApiResponse.FromCsxResponse(detectorResponse.Item1, detectorResponse.Item2));
         }
 
+        protected async Task<IActionResult> ListSystemInvokers(TResource resource)
+        {
+            DateTimeHelper.PrepareStartEndTimeWithTimeGrain(string.Empty, string.Empty, string.Empty, out DateTime startTimeUtc, out DateTime endTimeUtc, out TimeSpan timeGrainTimeSpan, out string errorMessage);
+            OperationContext<TResource> context = PrepareContext(resource, startTimeUtc, endTimeUtc);
+
+            await this._sourceWatcherService.Watcher.WaitForFirstCompletion();
+
+            var systemInvokers = _invokerCache.GetSystemInvokerList<TResource>(context)
+               .Select(p => new DiagnosticApiResponse { Metadata = p.EntryPointDefinitionAttribute });
+
+            return Ok(systemInvokers);
+        }
+
+        protected async Task<IActionResult> GetSystemInvoker(TResource resource, string invokerId, string detectorId, string startTime, string endTime, string timeGrain)
+        {
+            if (!DateTimeHelper.PrepareStartEndTimeWithTimeGrain(startTime, endTime, timeGrain, out DateTime startTimeUtc, out DateTime endTimeUtc, out TimeSpan timeGrainTimeSpan, out string errorMessage))
+            {
+                return BadRequest(errorMessage);
+            }
+
+            Dictionary<string, string> additionalContext = new Dictionary<string, string>
+            {
+                { "DetectorId", detectorId }
+            };
+
+            OperationContext<TResource> context = PrepareContext(resource, startTimeUtc, endTimeUtc, false, additionalContext);
+
+            await this._sourceWatcherService.Watcher.WaitForFirstCompletion();
+            var dataProviders = new DataProviders.DataProviders(_dataSourcesConfigService.Config);
+            var invoker = this._invokerCache.GetSystemInvoker<TResource>(invokerId, context);
+
+            if (invoker == null)
+            {
+                return null;
```

Comment:
return NotFound()

Reply:
I am using NotFound() for the final entityInvoker execution response here, and null for invokerCache response.